### ISSUE TITLE
Chrome standalone launcher fails

### ISSUE THREAD

**Issue by Bratzilla** [Wintermute0110/plugin.program.AEL#92]
I've been using AEL to create standalone browser launch instances for Chrome for a long while now, and this still works with my Kodi 17.6 installation (Windows). However, this does not work for Kodi 18.1, also on Windows. I've included the pertinent code from my Kodi log below.

FYI, the launcher is setup for Netflix, and is one I still successfully use on my Kodi 17.6 machine.

I noticed that Firefox generated a similar error as well, but unlike Chrome, the browser did launch. Please let me know if you want me to also provide the Firefox log. Thanks!

`019-02-26 01:05:28.234 T:8776   ERROR: AEL INFO : _run_standalone_launcher() categoryID 50419552627d6969b627b7cdb70b844d
2019-02-26 01:05:28.234 T:8776   ERROR: AEL INFO : _run_standalone_launcher() launcherID 208a86d88ebb2554ac265b291b20c690
2019-02-26 01:05:28.234 T:8776   ERROR: AEL INFO : _run_standalone_launcher() application   "C:\Program Files (x86)\Google\Chrome\Application\chrome.exe"
2019-02-26 01:05:28.234 T:8776   ERROR: AEL INFO : _run_standalone_launcher() apppath       "C:\Program Files (x86)\Google\Chrome\Application"
2019-02-26 01:05:28.234 T:8776   ERROR: AEL INFO : _run_standalone_launcher() app_basename  "chrome.exe"
2019-02-26 01:05:28.234 T:8776   ERROR: AEL INFO : _run_standalone_launcher() app_ext       ".exe"
2019-02-26 01:05:28.234 T:8776   ERROR: AEL INFO : _run_standalone_launcher() launcher name "Netflix"
2019-02-26 01:05:28.234 T:8776   ERROR: AEL INFO : _run_standalone_launcher() raw arguments   " --profile-directory="Profile 5" --app=https://netflix.com --fullscreen --start-maximized
"
2019-02-26 01:05:28.234 T:8776   ERROR: AEL INFO : _run_standalone_launcher() final arguments " --profile-directory="Profile 5" --app=https://netflix.com --fullscreen --start-maximized
"
2019-02-26 01:05:28.750 T:8776   ERROR: AEL ERROR: SingleInstance::__exit__() Unhandled excepcion in protected code
2019-02-26 01:05:28.756 T:8776   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.WindowsError'>
                                            Error Contents: [Error 740] The requested operation requires elevation
                                            Traceback (most recent call last):
                                              File "C:\Users\HomePC\AppData\Roaming\Kodi\addons\plugin.program.advanced.emulator.launcher\addon.py", line 39, in <module>
                                                main.Main().run_plugin()
                                              File "C:\Users\HomePC\AppData\Roaming\Kodi\addons\plugin.program.advanced.emulator.launcher\resources\main.py", line 270, in run_plugin
                                                self.run_protected(command, args)
                                              File "C:\Users\HomePC\AppData\Roaming\Kodi\addons\plugin.program.advanced.emulator.launcher\resources\main.py", line 371, in run_protected
                                                self._command_run_standalone_launcher(args['catID'][0], args['launID'][0])
                                              File "C:\Users\HomePC\AppData\Roaming\Kodi\addons\plugin.program.advanced.emulator.launcher\resources\main.py", line 7096, in _command_run_standalone_launcher
                                                self._run_process(application.getPath(), arguments, application.getDir(), app_ext, non_blocking_flag)
                                              File "C:\Users\HomePC\AppData\Roaming\Kodi\addons\plugin.program.advanced.emulator.launcher\resources\main.py", line 7466, in _run_process
                                                retcode = subprocess.call(exec_list, cwd = apppath.encode('utf-8'), close_fds = True)
                                              File "C:\Program Files\Kodi\system\python\Lib\subprocess.py", line 168, in call
                                                return Popen(*popenargs, **kwargs).wait()
                                              File "C:\Program Files\Kodi\system\python\Lib\subprocess.py", line 390, in __init__
                                                errread, errwrite)
                                              File "C:\Program Files\Kodi\system\python\Lib\subprocess.py", line 640, in _execute_child
                                                startupinfo)
                                            WindowsError: [Error 740] The requested operation requires elevation
                                            -->End of Python script error report<--
2`

**Comment by Wintermute0110** [comment]
Please put the full debug log on https://paste.kodi.tv/ I need all the debug information AEL produces. Thank you.

**Comment by Bratzilla** [comment]
> Please put the full debug log on https://paste.kodi.tv/ I need all the debug information AEL produces. Thank you.

Just got back in and saw your message. I will grab the full log and post as requested in just a bit. Thanks!

**Comment by Bratzilla** [comment]
Full Kodi log is now posted via the link you provided. This is my first time doing this, so if I've made any error or omission, please don't hesitate to let me know. Thanks again!

**Comment by Wintermute0110** [comment]
@Bratzilla I cannot see the log. To use hastebin, first open the log with a text editor. If using Windows I recommend Notepad++. Then, select all (Control+A), copy (Control+C), then go to your browser and paste the log context on hastebin. Then click on the Save button.

This will create an URL like https://paste.kodi.tv/oxoqoworad that I can use to read the log :)

**Comment by Bratzilla** [comment]
> @Bratzilla I cannot see the log. To use hastebin, first open the log with a text editor. If using Windows I recommend Notepad++. Then, select all (Control+A), copy (Control+C), then go to your browser and paste the log context on hastebin. Then click on the Save button.
> 
> This will create an URL like https://paste.kodi.tv/oxoqoworad that I can use to read the log :)

DO'OH! Trying again...

**Comment by Bratzilla** [comment]
Done! Below is the link that was generated:

https://paste.kodi.tv/qajulanuqu.kodi

**Comment by Wintermute0110** [comment]
Sorry for the late response.

Apparently, Chrome requires elevation to be run. This means that the user running Kodi does not have permissions to run the executable. Go to directory `C:\Program Files (x86)\Google\Chrome\Application` and right click on `chrome.exe`, check the properties of the file and make sure it can be run by a non-administrator user.

**Comment by Bratzilla** [comment]
> Sorry for the late response.
> 
> Apparently, Chrome requires elevation to be run. This means that the user running Kodi does not have permissions to run the executable. Go to directory `C:\Program Files (x86)\Google\Chrome\Application` and right click on `chrome.exe`, check the properties of the file and make sure it can be run by a non-administrator user.

Hello and thanks for the feedback! I tried your solution and it worked. I set chrome.exe to run as administrator and changed that permission for all users on my system (it's a home network so no worries there). To be safe, I also did the same to kodi.exe and firefox.exe. I am still testing, but so far, everything seems to now be running perfectly.

This issue should probably be brought to the attention of the larger community since it isn't plugin-specific. If I'd known this I could have resolved it myself. I'd searched but couldn't find any documentation.

Feel free to close if you wish.

Thanks again!
